### PR TITLE
Add Yotpo Destination to Segment with test action

### PR DESCRIPTION
**Pull request by vladimir-marinov-yotpo**
_A summary of your pull request, including the what change you're making and why._
This PR will add the initial version of the Yotpo destination. This is so we can set up the authentication process, and at a later date, actions will be added to destination.

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Comment by vladimir-marinov-yotpo** [comment]
@joe-ayoub-segment

File: packages/destination-actions/src/destinations/yotpo/index.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      })
+    },
+    refreshAccessToken: async (request, data) => {
+      const promise = await request<AccessTokenResponse>(`https://developers.yotpo.com/v2/oauth/token`, {
+        method: 'post',
+        json: {
+          client_id: data.auth.clientId,
+          client_secret: data.auth.clientSecret,
+          grant_type: 'authorization_code',
+          code: 'code', // TODO - this is the generated code from the authorization step, how can we get it or it would be added automatically by Segment?
```
Comment (vladimir-marinov-yotpo) [inline packages/destination-actions/src/destinations/yotpo/index.ts:38]: @joe-ayoub-segment Could you take a look at this todo? Is it okay to delete the code field?

File: packages/destination-actions/src/destinations/yotpo/index.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        type: 'string',
+        required: true
+      }
+    },
+    testAuthentication: (request, data) => {
+      return request(`https://developers.yotpo.com/v2/${data.settings.store_id}/info`, {
+        method: 'get'
+      })
+    },
+    refreshAccessToken: async (request, data) => {
```
Comment (vladimir-marinov-yotpo) [inline packages/destination-actions/src/destinations/yotpo/index.ts:31]: @joe-ayoub-segment Just to be on the safe side, we need this func even if yotpo tokens do not expire, right?

**Comment by joe-ayoub-segment** [comment]
Hi @vladimir-marinov-yotpo - I wasn't able to deploy the code as there were too many decisions which needed to be made in order for me to get the code working. 

I'd be happy to work through these on a call with you so I can explain the decisions and you can decide which way to go. 

Could you schedule a 30 minute call with me so we can get this done please? https://calendly.com/joe_ayoub/

**Comment by joe-ayoub-segment** [comment]
Closing this PR as code moved to internal PR: https://github.com/segmentio/action-destinations/pull/1932

**Comment by joe-ayoub-segment** [comment]
Hi @vladimir-marinov-yotpo - PR deployed. I'll share next steps via email

**Pull request closed without merging**